### DISCUSSION TITLE
"LOG" function to store position of a successful straight probe in a file

### DISCUSSION THREAD

**Discussion by JPS31F** [winder/Universal-G-Code-Sender#2656] [Help]
I would like to "scan"a 3D surface and need to save successful straight probes during a gcode processing. Is ther a way to save the location of the probe ? As LOGOPEN in linuxcnc during a G38 whichjstore LOG,#5061 #5062 #5063 ?
OI am using a Genmitsu 4030 PROVERXL v2 CNC and UGS version 2.0.14
Thanks for your help

**Comment by breiler** [comment]
The currently only option is to upgrade to the latest version of UGS (2.1.9) and use the AutoLevel plugin. This will allow you to probe a grid and export the probed points to a text file.

There is an issue in 2.1.9 where controllers with low precisions will report coordinates with rounding errors which will make the plugin throw an error. This has been fixed in 2.1.11, however it is still in under testing: https://github.com/winder/Universal-G-Code-Sender/releases/tag/v2.1.11

**Reply by JPS31F** [reply]
Thank you for such a fast answer !!! I shall have a look.. I also noticed a plugin named Surface scanner ??? but could not find how to start it...

**Reply by breiler** [reply]
I think that "Surface scanner" was the previous name for the Autoleveler.
